File: codegen/results/v002/qwen3.5_27b-nvfp4_nothink/20260412_033843/typescript/008_prime_numbers/solution.ts
#!/usr/bin/env tsx

/**
 * Script: Find all prime numbers up to N (inclusive) using the Sieve of Eratosthenes.
 * Input: A single integer argument from command line (process.argv[2]).
 * Output: Prints each prime number on a new line to stdout.
 */

function main(): void {
  const input = process.argv[2];

  if (input === undefined) {
    return;
  }

  const n = parseInt(input, 10);

  // If N is less than 2, there are no primes, so print nothing and exit.
  if (isNaN(n) || n < 2) {
    return;
  }

  // Sieve of Eratosthenes to find all primes up to n
  // Create a boolean array "isPrime" where index represents the number.
  // Initialize all entries as true.
  const isPrime = new Array<boolean>(n + 1).fill(true);

  // 0 and 1 are not prime numbers
  isPrime[0] = false;
  isPrime[1] = false;

  // Only need to check up to sqrt(n)
  for (let i = 2; i * i <= n; i++) {
    if (isPrime[i]) {
      // If i is prime, then mark all multiples of i as non-prime
      // Start from i*i because smaller multiples would have been handled already
      for (let j = i * i; j <= n; j += i) {
        isPrime[j] = false;
      }
    }
  }

  // Collect and print all prime numbers
  const primes: string[] = [];
  for (let i = 2; i <= n; i++) {
    if (isPrime[i]) {
      primes.push(i.toString());
    }
  }

  // Print all found primes, one per line
  if (primes.length > 0) {
    console.log(primes.join('\n'));
  }
}

main();

File: codegen/results/v002/qwen3.5_27b-nvfp4_nothink/20260412_033843/python/008_prime_numbers/solution.py
import sys

def is_prime(num):
    if num < 2:
        return False
    if num == 2:
        return True
    if num % 2 == 0:
        return False
    
    # Check for factors from 3 up to the square root of num
    limit = int(num ** 0.5) + 1
    for i in range(3, limit, 2):
        if num % i == 0:
            return False
    return True

def main():
    # Accept a single integer argument N from the command line
    if len(sys.argv) != 2:
        return
    
    try:
        n = int(sys.argv[1])
    except ValueError:
        return

    # Print all prime numbers up to and including N, one per line
    for number in range(2, n + 1):
        if is_prime(number):
            print(number)

if __name__ == "__main__":
    main()